packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -109,8 +115,8 @@ export class AwsCliCompatible {
       : nodeProviderChain;
   }
 
-  public static requestHandlerBuilder(options: SdkHttpOptions = {}): NodeHttpHandlerOptions {
-    const agent = this.proxyAgent(options);
+  public async requestHandlerBuilder(options: SdkHttpOptions = {}): Promise<NodeHttpHandlerOptions> {
+    const agent = await new ProxyAgentProvider(this.ioHelper).create(options);
```

Comment:
Since we are changing these things, it would be better to pass then agent as a constructor argument. Let the SDK Provider, which is the factory class, to also instantiate this.

Reply:
I tried, but it has a bunch of knock-on effects that I don't want to deal with right now.